e2e/testcafe-devextreme/tests/cardView/columnChooser/functional.ts

```diff
@@ -0,0 +1,106 @@
+import CardView from 'devextreme-testcafe-models/cardView';
+import url from '../../../helpers/getPageUrl';
+import { createWidget } from '../../../helpers/createWidget';
+
+fixture.disablePageReloads`CardView - ColumnChooser.Functional`
+  .page(url(__dirname, '../../container.html'));
+
+function testsFactory(testModel: {
+  name: string;
+  hideFirstColumn: (t: TestController, cardView: CardView) => Promise<void>;
+  showFirstColumn: (t: TestController, cardView: CardView) => Promise<void>;
+  assertFirstColumnVisible: (t: TestController, cardView: CardView) => Promise<void>;
+  assertFirstColumnHidden: (t: TestController, cardView: CardView) => Promise<void>;
+  config: any; // TODO: add typing
+}) {
+  test(`column chooser in ${testModel.name} mode should work after multiple hide/show actions`, async (t) => {
+    const cardView = new CardView('#container');
+
+    await cardView.apiShowColumnChooser();
+
+    await testModel.hideFirstColumn(t, cardView);
+    await testModel.assertFirstColumnHidden(t, cardView);
```

Comment:
Minor: I think it's better to check the visibility of all columns:

expect(columnsVisibility).eql([false, true, true])

or something like:

expect(headerPanel.text).eql('b c');